src/datasets/load.py

```diff
@@ -88,6 +89,45 @@
 ALL_ALLOWED_EXTENSIONS = list(_EXTENSION_TO_MODULE.keys()) + [".zip"]
 
 
+def _raise_timeout_error(signum, frame):
+    raise ValueError(
+        "Loading this dataset requires you to execute custom code contained in the dataset repository on your local "
+        "machine. Please set the option `trust_remote_code=True` to permit loading of this dataset."
+    )
+
+
+def resolve_trust_remote_code(trust_remote_code: Optional[bool], repo_id: str) -> bool:
```

Comment:
It complicates the logic, but Transformers does the same, so it's okay. 

A comment to the `transformers` version would be nice (to make it clear why it's implemented as-is)